connector-runtime/spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/inbound/WebhookControllerTestZeebeTests.java

```diff
@@ -205,6 +212,44 @@ public void testSuccessfulProcessingWithFailedActivation() throws Exception {
 
     assertEquals(200, responseEntity.getStatusCode().value());
     assertFalse(Objects.requireNonNull(responseEntity.getBody()).isActivated());
+    assertFalse(responseEntity.getBody().getResponseData().isPresent());
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testSuccessfulProcessingWithActivationCorrelationHidden() throws Exception {
+    WebhookConnectorExecutable webhookConnectorExecutable = mock(WebhookConnectorExecutable.class);
+    WebhookResult webhookResult = mock(WebhookResult.class);
+    when(webhookResult.request()).thenReturn(new MappedHttpRequest(Map.of(), Map.of(), Map.of()));
+    // default use-case, when result expression not set
+    when(webhookResult.responseBodyExpression()).thenReturn(webhookResultContext -> null);
+    when(webhookConnectorExecutable.triggerWebhook(any(WebhookProcessingPayload.class)))
+            .thenReturn(webhookResult);
+
+    var webhookDef = webhookDefinition("processA", 1, "myPath");
+    var webhookContext =
+            new InboundConnectorContextImpl(
+                    secretProvider, v -> {}, webhookDef, correlationHandler, (e) -> {}, mapper);
+
+    // Register webhook function 'implementation'
+    webhookConnectorRegistry.register(
+            new ActiveInboundConnector(webhookConnectorExecutable, webhookContext));
+
+    deployProcess("processA");
+
+    ResponseEntity<Map> responseEntity =
+            (ResponseEntity<Map>)
+                    controller.inbound(
+                            "myPath",
+                            new HashMap<>(),
+                            "{}".getBytes(),
+                            new HashMap<>(),
+                            new MockHttpServletRequest());
+
+    assertEquals(200, responseEntity.getStatusCode().value());
+
+    var result = responseEntity.getBody();
+    assertNull(result);
```

Comment:
Would it make sense to json assert the response?

Reply:
it's null :)

as per contract - if no response body expression given, we just return `200 OK` and no body. Json assert won't be different here.

Reply:
Right, I meant to add a case where its not null.

Reply:
I see your point. For that case we do parsing already, like:

```
ResponseEntity<InboundConnectorResult<?>> responseEntity =
        (ResponseEntity<InboundConnectorResult<?>>)
            controller.inbound(
                "myPath",
                new HashMap<>(),
                "{}".getBytes(),
                new HashMap<>(),
                new MockHttpServletRequest());
```

And I added those use-cases as asserts by actual object properties.

Reply:
I meant a real test against the actual rendered json response and not the `ResponseEntity`.

Reply:
But we don't have these now. Or at least I am not aware of. If so, I would suggest to have a separate task that I can take after I am back. Today is the last day before I leave for FTO, and there are more urgent tasks left - regression testing and documentation.

Reply:
Alright.